realm/realm-library/src/main/java/io/realm/RealmSet.java

```diff
@@ -0,0 +1,389 @@
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isManaged() {
+        return setStrategy.isManaged();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isValid() {
+        return setStrategy.isValid();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isFrozen() {
+        return setStrategy.isFrozen();
+    }
+
+    // ------------------------------------------
+    // Set API
+    // ------------------------------------------
+
+    @Override
+    public int size() {
+        return setStrategy.size();
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return setStrategy.isEmpty();
+    }
+
+    @Override
+    public boolean contains(@Nullable Object o) {
+        return setStrategy.contains(o);
+    }
+
+    @NotNull
+    @Override
+    public Iterator<E> iterator() {
+        return setStrategy.iterator();
+    }
+
+    @NotNull
+    @Override
+    public Object[] toArray() {
+        return setStrategy.toArray();
+    }
+
+    @NotNull
+    @Override
+    public <T> T[] toArray(@NotNull T[] a) {
+        return setStrategy.toArray(a);
+    }
+
+    @Override
+    public boolean add(@Nullable E e) {
+        return setStrategy.add(e);
+    }
+
+    @Override
+    public boolean remove(@Nullable Object o) {
+        return setStrategy.remove(o);
+    }
+
+    @Override
+    public boolean containsAll(@NotNull Collection<?> c) {
+        return setStrategy.containsAll(c);
+    }
+
+    @Override
+    public boolean addAll(@NotNull Collection<? extends E> c) {
+        return setStrategy.addAll(c);
+    }
+
+    @Override
+    public boolean retainAll(@NotNull Collection<?> c) {
+        return setStrategy.retainAll(c);
+    }
+
+    @Override
+    public boolean removeAll(@NotNull Collection<?> c) {
+        return setStrategy.removeAll(c);
+    }
+
+    @Override
+    public void clear() {
+        setStrategy.clear();
+    }
+
+    // ------------------------------------------
+    // Freezable API
+    // ------------------------------------------
+
+    @Override
+    public RealmSet<E> freeze() {
+        return setStrategy.freeze();
+    }
+
+    /**
+     * TODO
+     *
+     * @param <E>
+     */
+    private abstract static class SetStrategy<E> implements Set<E>, ManageableObject, Freezable<RealmSet<E>> {
+        // TODO
+    }
+
+    /**
+     * TODO
+     *
+     * @param <E>
+     */
+    private static class ManagedSetStrategy<E> extends SetStrategy<E> {
+
+        private final ManagedSetManager<E> managedSetManager;
+
+        private ManagedSetStrategy(ManagedSetManager<E> managedSetManager) {
+            this.managedSetManager = managedSetManager;
+        }
+
+        // ------------------------------------------
+        // ManageableObject API
+        // ------------------------------------------
+
+        @Override
+        public boolean isManaged() {
+            return false;
```

Comment:
Maybe not relevant now, but I would advise to `throw UnsupportedOperation("Not implemented yet")` instead of all the incorrect return values. It avoids you being tricked by missing to update one of the implementations later.

Reply:
Yeah, that's a ver good point. Too late for this feature though because all the upcoming PRs have dramatic changes in the written tests, but I'll remember for next time.